testsuites/syncgateway/upgrade/conftest.py

```diff
@@ -0,0 +1,354 @@
+""" Setup for Sync Gateway functional tests """
```

Comment:
As it is hard to maintain lot of conftest files. is it something you can put upgrade test in same folder as system test and use it?

Reply:
The conftest for upgrade is very different from that of the system test, like, when provisioning happens, --xattrs won't even take effect. There are other such differences. Hence, cannot combine it with other conftest.py

Reply:
can you add  common conftest for systemTest, upgrade and any other tests and see if it works?

Reply:
I looked at it again. The parameters are very different. I don't see a way to combine the conftests. The granular differences are enough to keep it separate. The high level intent is the same, so it should not be too difficult to maintain.